### ISSUE TITLE
NPE when deleting message

### ISSUE THREAD

**Issue by deepthidevaki** [camunda-cloud/zeebe#7988]
**Describe the bug**

```

java.lang.RuntimeException: Unexpected error occurred during zeebe db transaction operation.

        at io.camunda.zeebe.db.impl.rocksdb.transaction.DefaultTransactionContext.runInTransaction (DefaultTransactionContext.java:46)
        at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.ensureInOpenTransaction (TransactionalColumnFamily.java:52)
        at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.lambda$whileEqualPrefix$10 (TransactionalColumnFamily.java:269)
        at io.camunda.zeebe.db.impl.rocksdb.transaction.ColumnFamilyContext.withPrefixKey (ColumnFamilyContext.java:111)
        at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.whileEqualPrefix (TransactionalColumnFamily.java:266)
        at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.whileEqualPrefix (TransactionalColumnFamily.java:248)
        at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.whileTrue (TransactionalColumnFamily.java:180)
        at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.whileTrue (TransactionalColumnFamily.java:112)
        at io.camunda.zeebe.engine.state.message.DbMessageState.visitMessagesWithDeadlineBefore (DbMessageState.java:303)
        at io.camunda.zeebe.engine.processing.message.MessageTimeToLiveChecker.run (MessageTimeToLiveChecker.java:32)
        at io.camunda.zeebe.util.sched.ActorJob.invoke (ActorJob.java:76)
        at io.camunda.zeebe.util.sched.ActorJob.execute (ActorJob.java:39)
        at io.camunda.zeebe.util.sched.ActorTask.execute (ActorTask.java:122)
        at io.camunda.zeebe.util.sched.ActorThread.executeCurrentTask (ActorThread.java:95)
        at io.camunda.zeebe.util.sched.ActorThread.doWork (ActorThread.java:78)
        at io.camunda.zeebe.util.sched.ActorThread.run (ActorThread.java:192)

Caused by: java.lang.NullPointerException

        at io.camunda.zeebe.engine.processing.message.MessageTimeToLiveChecker.writeDeleteMessageCommand (MessageTimeToLiveChecker.java:37)
        at io.camunda.zeebe.engine.state.message.DbMessageState.lambda$visitMessagesWithDeadlineBefore$1 (DbMessageState.java:309)
        at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.visit (TransactionalColumnFamily.java:310)
        at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.lambda$whileEqualPrefix$9 (TransactionalColumnFamily.java:289)
        at io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.lambda$ensureInOpenTransaction$0 (TransactionalColumnFamily.java:53)
        at io.camunda.zeebe.db.impl.rocksdb.transaction.DefaultTransactionContext.runInNewTransaction (DefaultTransactionContext.java:62)
        at io.camunda.zeebe.db.impl.rocksdb.transaction.DefaultTransactionContext.runInTransaction (DefaultTransactionContext.java:34)

```

As a result StreamProcessor went into "failed" state. One partition is unhealthy. Not sure if it will recover after a leader change.

![image](https://user-images.githubusercontent.com/1997478/137312885-4dd12191-3ccf-4701-802c-ade1b6a0a60e.png)

**To Reproduce**

Don't know.

**Expected behavior**

* No null pointer exception. 
* If it happens, StreamProcessor should  handle it (blacklisting the instance for eg:-) and continue processing other instances.

**Log/Stacktrace**

[Logs](https://console.cloud.google.com/errors/CIKS4a-D5dfqfA?service=zeebe&time=P7D&refresh=off&project=camunda-cloud-240911) 

**Environment:**
 - Camunda Cloud
 - Zeebe Version: 1.2.0

**Comment by npepinpe** [comment]
Partition will crash and not recover - assumption is even on leader change, the new leader will run into the same issue and also fail. In CCSaaS, as we have 2 partitions, it means half of them are unavailable. Worse if it's your deployment partition, then you can't deploy anything. So let's work on it soon.

**Comment by menski** [comment]
We have a support case about this https://jira.camunda.com/browse/SUPPORT-11962, it's a trial cluster which the customer uses as staging env

**Comment by menski** [comment]
For context, I tried to recover by deleting the leader pod and forcing a new leader. but the new leader triggered the same exception and ended up with an unhealthy partition 1, it seems that the state is in an inconsistent state.

**Comment by pihme** [comment]
The first cluster was already deleted. We might get hold of the second one, but it is unclear whether it contains the same error

**Comment by pihme** [comment]
So there are about 20 messages at the beginning of the iteration (assuming the iteration is ordered by deadline, that means it is the 20 oldest messages that can no longer be found)

```
| messageKey       | deadline      |
------------------------------------
| 2251799818453665 | 1634196457943 |
| 2251799818453672 | 1634196457979 |
| 2251799818453694 | 1634196462649 |
| 2251799818453695 | 1634196462658 |
```

The messages that can be found look all very similar to this:
```
Found message{"messageKey":2251799818454068,"message":{"name":"vantage-transcode-progress-update","correlationKey":"1ad52444-6ec1-4cb8-b919-5a9cd17dcd67","timeToLive":3600000,"variables":"gbBwZXJjZW50Q29tcGxldGVkojI3","messageId":"","deadline":1634196780412}}
```

They have no message ID, but other then that it looks very normal.

I have a branch based off of zdb where this is a little easier to debug.

**Comment by saig0** [comment]
> Is this expected? I know -1 is our default for no parent, but then I would not expect it to show up in that column family.

Yes. Currently, this is expected. Even if the element instance has no parent, it creates an entry in the column family (that is probably not really necessary).

**Comment by pihme** [comment]
Fixed with #8062 in `develop` and #8101 in `stable/1.2`
